Kafka cancelled exception not suppressed

Not sure if this is a recurrence of https://github.com/hazelcast/hazelcast-jet/issues/824

Stopping a job with `job.cancel()` throws an Interrupted Exception from Kafka 2.0.0,
using hazelcast-jet-kafka-0.7-20180919.141941-164 snapshot build

Job is doing
```
		StreamStage<Entry<Object, Object>> source = 
				pipeline.drawFrom(KafkaSources.kafka(properties, "neil"))

```

Gives this

```
23:21:54.290 [hz._hzInstance_1_oktobap.jet.blocking.ProcessorTasklet{kafkaSource#2}] WARN  com.hazelcast.jet.impl.execution.TaskletExecutionService - [127.0.0.1]:5701 [oktobap] [0.7-SNAPSHOT] Exception in ProcessorTasklet{kafkaSource#2}
 org.apache.kafka.common.errors.InterruptException: java.lang.InterruptedException
	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.maybeThrowInterruptException(ConsumerNetworkClient.java:492) ~[kafka-clients-1.0.2.jar!/:?]
	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:274) ~[kafka-clients-1.0.2.jar!/:?]
	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:230) ~[kafka-clients-1.0.2.jar!/:?]
	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:206) ~[kafka-clients-1.0.2.jar!/:?]
	at org.apache.kafka.clients.consumer.internals.ConsumerCoordinator.commitOffsetsSync(ConsumerCoordinator.java:601) ~[kafka-clients-1.0.2.jar!/:?]
	at org.apache.kafka.clients.consumer.KafkaConsumer.commitSync(KafkaConsumer.java:1211) ~[kafka-clients-1.0.2.jar!/:?]
	at com.hazelcast.jet.kafka.impl.StreamKafkaP.complete(StreamKafkaP.java:200) ~[hazelcast-jet-kafka-0.7-SNAPSHOT.jar!/:0.7-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:343) ~[hazelcast-jet-0.7-SNAPSHOT.jar!/:0.7-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:212) ~[hazelcast-jet-0.7-SNAPSHOT.jar!/:0.7-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:205) ~[hazelcast-jet-0.7-SNAPSHOT.jar!/:0.7-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$BlockingWorker.run(TaskletExecutionService.java:235) [hazelcast-jet-0.7-SNAPSHOT.jar!/:0.7-SNAPSHOT]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_181]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_181]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_181]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_181]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_181]
Caused by: java.lang.InterruptedException
	... 16 more
23:21:54.360 [hz._hzInstance_1_oktobap.async.thread-4] INFO  com.hazelcast.jet.impl.MasterContext - [127.0.0.1]:5701 [oktobap] [0.7-SNAPSHOT] Execution of job 'GPS Load', execution 4c4f-9dd3-6bb7-6fbe completed in 4,095 ms, reason=java.util.concurrent.CancellationException
```

--- a/hazelcast-jet-kafka/src/main/java/com/hazelcast/jet/kafka/impl/StreamKafkaP.java
+++ b/hazelcast-jet-kafka/src/main/java/com/hazelcast/jet/kafka/impl/StreamKafkaP.java
@@ -31,7 +31,6 @@ import org.apache.kafka.clients.consumer.ConsumerRecord;
 import org.apache.kafka.clients.consumer.ConsumerRecords;
 import org.apache.kafka.clients.consumer.KafkaConsumer;
 import org.apache.kafka.common.TopicPartition;
-import org.apache.kafka.common.errors.InterruptException;
 
 import javax.annotation.Nonnull;
 import java.util.Arrays;
@@ -171,34 +170,32 @@ public final class StreamKafkaP<K, V, T> extends AbstractProcessor {
             return false;
         }
 
-        ConsumerRecords<K, V> records = null;
         try {
+            ConsumerRecords<K, V> records = null;
             assignPartitions(true);
             if (!currentAssignment.isEmpty()) {
                 records = consumer.poll(POLL_TIMEOUT_MS);
             }
-        } catch (InterruptException e) {
-            // note this is Kafka's exception, not Java's
-            Thread.currentThread().interrupt();
+
+            traverser = isEmpty(records)
+                    ? watermarkSourceUtil.handleNoEvent()
+                    : traverseIterable(records).flatMap(record -> {
+                        offsets.get(record.topic())[record.partition()] = record.offset();
+                        T projectedRecord = projectionFn.apply(record);
+                        if (projectedRecord == null) {
+                            return Traversers.empty();
+                        }
+                        TopicPartition topicPartition = new TopicPartition(record.topic(), record.partition());
+                        return watermarkSourceUtil.handleEvent(projectedRecord, currentAssignment.get(topicPartition));
+                    });
+
+            emitFromTraverser(traverser);
+
+            if (!snapshottingEnabled) {
+                consumer.commitSync();
+            }
+        } catch (org.apache.kafka.common.errors.InterruptException e) {
             return false;
-        }
-
-        traverser = isEmpty(records)
-                ? watermarkSourceUtil.handleNoEvent()
-                : traverseIterable(records).flatMap(record -> {
-                    offsets.get(record.topic())[record.partition()] = record.offset();
-                    T projectedRecord = projectionFn.apply(record);
-                    if (projectedRecord == null) {
-                        return Traversers.empty();
-                    }
-                    TopicPartition topicPartition = new TopicPartition(record.topic(), record.partition());
-                    return watermarkSourceUtil.handleEvent(projectedRecord, currentAssignment.get(topicPartition));
-                });
-
-        emitFromTraverser(traverser);
-
-        if (!snapshottingEnabled) {
-            consumer.commitSync();
         }
 
         return false;
@@ -209,7 +206,7 @@ public final class StreamKafkaP<K, V, T> extends AbstractProcessor {
         if (consumer != null) {
             try {
                 consumer.close();
-            } catch (InterruptException ignored) {
+            } catch (org.apache.kafka.common.errors.InterruptException ignored) {
             }
         }
     }

